src/librustc_codegen_ssa/debuginfo/type_names.rs

```diff
@@ -190,11 +190,17 @@ pub fn push_debuginfo_type_name<'tcx>(
             // processing
             visited.remove(t);
         },
-        ty::Closure(..) => {
-            output.push_str("closure");
+        ty::Closure(def_id, ..) => {
+            output.push_str(&format!(
+                "closure-{}",
```

Comment:
I would've suggested the same syntax as `rustc-demangle`, but this is probably fine.
This entire module seems a bit redundant though, but I don't think I have the time to refactor it post-`ty::print`.

Reply:
You could open an issue and give some mentors.

Reply:
Yeah but that would require me to actually look at what this code is doing.
There might already be an issue open for "things to clean up post-`ty::print`".